src/main/java/org/jenkinsci/plugins/mesos/MesosCleanupThread.java

```diff
@@ -0,0 +1,110 @@
+import hudson.model.TaskListener;
+import jenkins.model.Jenkins;
+
+/**
+ This file is part of the JCloud Jenkins Plugin. (https://github.com/jenkinsci/jclouds-plugin)
+ This file was modified to work with the Mesos Jenkins plugin and based off the JClouds Jenkins Plugin.
+
+ According to the Jenkins Plugin guide (https://wiki.jenkins-ci.org/display/JENKINS/Before+starting+a+new+plugin),
+ If no license is defined in the form of a header, LICENSE file, or in the license section, the code is assumed to be
+ under The MIT License.
+
+ The MIT License (MIT)
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+@Extension
+public class MesosCleanupThread extends AsyncPeriodicWork {
+
+    public MesosCleanupThread() {
+        super("Mesos pending deletion slave cleanup");
+    }
+
+    @Override
+    public long getRecurrencePeriod() {
+        return MIN * 5;
+    }
+
+    public static void invoke() {
+        getInstance().run();
+    }
+
+    private static MesosCleanupThread getInstance() {
+        return Jenkins.getInstance().getExtensionList(AsyncPeriodicWork.class).get(MesosCleanupThread.class);
+    }
+
+    @Override
+    protected void execute(TaskListener listener) {
+        final ImmutableList.Builder<ListenableFuture<?>> deletedNodesBuilder = ImmutableList.<ListenableFuture<?>>builder();
+        ListeningExecutorService executor = MoreExecutors.listeningDecorator(Computer.threadPoolForRemoting);
+        final ImmutableList.Builder<MesosComputer> computersToDeleteBuilder = ImmutableList.<MesosComputer>builder();
+
+        for (final Computer c : Jenkins.getInstance().getComputers()) {
+            if (MesosComputer.class.isInstance(c)) {
+                if (((MesosComputer) c).getNode().isPendingDelete()) {
+                    final MesosComputer comp = (MesosComputer) c;
+                    computersToDeleteBuilder.add(comp);
+                    logger.log(Level.INFO, "Marked " + comp.getName() + " for deletion");
```

Comment:
nit: String.format("Marked %s for deletion.", comp.getName())

The same for other string formatting sections.